crates/biome_formatter/src/trivia.rs

```diff
@@ -117,7 +160,25 @@ where
                     [
                         line_suffix(&format_with(|f| {
                             match comment.lines_before() {
-                                0 | 1 => write!(f, [hard_line_break()])?,
+                                _ if should_nestle => {}
```

Comment:
This is the biggest change from what existed before. We used to say that all trailing comments got put on their own lines. This now says that comments starting on the same line as the previous comment will always stay on that same line, and will just have a `space` if they aren't nestled.

afaict, that matches Prettier's behavior in all cases, and all of the tests still pass, so I assume that's true.